qemu/exec.c

```diff
@@ -1484,14 +1489,17 @@ static void tcg_commit(MemoryListener *listener)
 
 void address_space_init_dispatch(AddressSpace *as)
 {
-    as->dispatch = NULL;
-    as->dispatch_listener = (MemoryListener) {
-        .begin = mem_begin,
-        .commit = mem_commit,
-        .region_add = mem_add,
-        .region_nop = mem_add,
-        .priority = 0,
+    MemoryListener ml = {
+        mem_begin,
+        mem_commit,
+        mem_add,
+        NULL,
+        mem_add,
+        NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,
+        0,
     };
+    as->dispatch = NULL;
+    as->dispatch_listener = ml;
     memory_listener_register(as->uc, &as->dispatch_listener, as);
```

Comment:
yes i saw what you did. totally agreed that must be ugly in the cases you mentioned.

but like in the code above we can avoid that.